chart/templates/postgres-monitor.yaml

```diff
@@ -0,0 +1,18 @@
+{{- if .Values.monitoring }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: postgres-exporter
+  namespace: postgres
+  annotations:
```

Comment:
related to above, it seems you should be able to configure this although there was kind of funky behavior with the initial release of the podmonitor generator and mutations in uds-core so this may be able to be put back in the package CR once the next release > 0.24.0 is out.

Reply:
I'll be happy to refactor once the functional is in the package CR and good to go.

Reply:
0.24.1 is out as of today

Reply:
I attempted to use the monitor key of the Package CR today on 24.1 to see if I could just setup the PodMonitor there and it seems prometheus is still unable to retrieve data from the endpoint. I see all the way to the ISTIO proxy on the PG Pod that the traffic is there and it is allowed as "InboundPassthroughClusterIpv4". The only way I've been able to get this podmoniotr working is with the MTLS exception for the port.

Reply:
Just putting this here since it worked for me for the podmonitor w/o the peerauth exception:

```yaml
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: postgres-exporter
  namespace: postgres
spec:
  scrapeClass: istio-certs
  namespaceSelector:
    matchNames:
      - postgres
  podMetricsEndpoints:
    - port: exporter
      scheme: https
      enableHttp2: false
  selector:
    matchLabels:
      application: spilo
```

Reply:
Also would prefer to default to monitoring being on (potentially removing the configurability entirely to just go with a pattern more similar to the gitlab package - unless there is something I am missing)

Reply:
@Racer159 Thanks, there is definatly some things in here I didn't try. Thanks for sharing I'll refactor the PR. 

As for defaulting to off I was just erroring on the side of caution and avoiding inducing a change if people are using it. I'm happy to just default to the end point is there and exposed. I'll refactor that as well.